### PR TITLE
refactor: make workflow generic

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
Refactors retrievers and agent to be framework agnostic.

### CODE REVIEW

File: ml_frameworks_bot/retrieval/heuristic_retriever.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            self._api_to_doc_mapping = json.load(file)
+        if repository_local_path is None:
+            self.repository_local_path = get_wandb_artifact(
+                artifact_name=RepositoryMapping[framework]["artifact_address"],
+                artifact_type="docs",
+            )
+        else:
+            self.repository_local_path = repository_local_path
+
+    def load_doc(self, op: str) -> BaseNode:
```
Comment (soumik12345) [inline ml_frameworks_bot/retrieval/heuristic_retriever.py:35]: Decorate this function with `@weave.op()`.

File: ml_frameworks_bot/retrieval/neural_retriever.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        return_nodes: bool = True,
+        vector_index_persist_dir: Optional[str] = None,
+        artifact_name: Optional[str] = None,
+        artifact_aliases: Optional[List[str]] = [],
+        track_load_documents: bool = False,
+    ) -> VectorStoreIndex:
+        if self.repository_local_path is not None:
+            # Load documents as nodes
+            load_document_fn = (
+                weave.op()(self.load_documents)
```
Comment (soumik12345) [inline ml_frameworks_bot/retrieval/neural_retriever.py]: No need to make document loading into ops.

File: ml_frameworks_bot/retrieval/neural_retriever.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+                    )
+        return document_nodes
+
+    def index_documents(
+        self,
+        return_nodes: bool = True,
+        vector_index_persist_dir: Optional[str] = None,
+        artifact_name: Optional[str] = None,
+        artifact_aliases: Optional[List[str]] = [],
+        track_load_documents: bool = False,
```
Comment (soumik12345) [inline ml_frameworks_bot/retrieval/neural_retriever.py]: Same, we don't need to make document loading into ops.

File: ml_frameworks_bot/translation_agent.py [hunk 4]
```diff
@@ -5,23 +5,21 @@
...
-    keras_op: str
+
+class OpWithAPIReference(BaseModel):
+    op: str
     api_reference: str
     api_reference_path: str
 
 
-class KerasDocumentationAgent(weave.Model):
+class TranslationAgent(weave.Model):
```
Comment (soumik12345) [inline ml_frameworks_bot/translation_agent.py]: Let's rename this to `OpExtractor`

**Review by soumik12345 — CHANGES_REQUESTED** [review]

File: ml_frameworks_bot/translation_agent.py (hunk 4 above)
Comment (SauravMaheshkar) [reply ml_frameworks_bot/translation_agent.py]: Agreed 😄

**Review by soumik12345 — APPROVED** [review]

**Pull request merged**